### PR TITLE
Conversion PearDatabase to yii Users_Record_Model

### PR DESCRIPTION
**Pull request by mlorencik**

### CODE REVIEW

File: modules/Users/models/Record.php [hunk 1]
```diff
@@ -721,32 +727,23 @@ public function delete()
...
 		$users = [];
-		if ($noOfUsers > 0) {
-			for ($i = 0; $i < $noOfUsers; ++$i) {
-				$userId = $db->queryResult($result, $i, 'id');
-				$userModel = self::getInstanceFromUserObject(\App\User::getUserModel($userId));
-				$users[$userModel->getId()] = $userModel;
-			}
+		while ($row = $dataReader->read()) {
+			$userModel = self::getInstanceFromUserObject(\App\User::getUserModel($row['id']));
+			$users[$userModel->getId()] = $userModel;
```
Comment (mariuszkrzaczkowski) [inline modules/Users/models/Record.php]: $users[$id] = self::getInstanceFromUserObject(\App\User::getUserModel($id));

File: modules/Users/models/Record.php [hunk 2]
```diff
@@ -721,32 +727,23 @@ public function delete()
...
-		$noOfUsers = $db->numRows($result);
+		$dataReader = (new App\Db\Query())->select('id')->from('vtiger_users')->where(['status' => 'ACTIVE', 'is_admin' => 'on'])->createCommand()->query();
 		$users = [];
-		if ($noOfUsers > 0) {
-			for ($i = 0; $i < $noOfUsers; ++$i) {
-				$userId = $db->queryResult($result, $i, 'id');
-				$userModel = self::getInstanceFromUserObject(\App\User::getUserModel($userId));
-				$users[$userModel->getId()] = $userModel;
-			}
+		while ($row = $dataReader->read()) {
```
Comment (mariuszkrzaczkowski) [inline modules/Users/models/Record.php]:
```suggestion
		while ($id = $dataReader->readColumn(0)) {
```

File: modules/Users/models/Record.php [hunk 3]
```diff
@@ -721,32 +727,23 @@ public function delete()
 
 	public function getActiveAdminUsers()
 	{
-		$db = PearDatabase::getInstance();
-
-		$sql = 'SELECT id FROM vtiger_users WHERE status=? && is_admin=?';
-		$result = $db->pquery($sql, ['ACTIVE', 'on']);
-
-		$noOfUsers = $db->numRows($result);
+		$dataReader = (new App\Db\Query())->select('id')->from('vtiger_users')->where(['status' => 'ACTIVE', 'is_admin' => 'on'])->createCommand()->query();
```
Comment (mariuszkrzaczkowski) [inline modules/Users/models/Record.php]:
```suggestion
		$dataReader = (new App\Db\Query())->select(['id'])->from('vtiger_users')->where(['status' => 'ACTIVE', 'is_admin' => 'on'])->createCommand()->query();
```

File: modules/Users/models/Record.php [hunk 4]
```diff
@@ -721,32 +727,22 @@ public function delete()
 
 	public function getActiveAdminUsers()
 	{
-		$db = PearDatabase::getInstance();
-
-		$sql = 'SELECT id FROM vtiger_users WHERE status=? && is_admin=?';
-		$result = $db->pquery($sql, ['ACTIVE', 'on']);
-
-		$noOfUsers = $db->numRows($result);
+		$dataReader = (new App\Db\Query())->select(['id'])->from('vtiger_users')->where(['status' => 'ACTIVE', 'is_admin' => 'on'])->createCommand()->query();
```
Comment (rskrzypczak) [inline modules/Users/models/Record.php]:
```suggestion
		$dataReader = (new App\Db\Query())->select(['id'])->from('vtiger_users')->where(['status' => 'Active', 'is_admin' => 'on'])->createCommand()->query();
```

**Pull request merged**